lib/if.c

```diff
@@ -654,7 +654,11 @@ static void if_set_name(struct interface *ifp, const char *name)
 /* Does interface up ? */
 int if_is_up(const struct interface *ifp)
```

Comment:
This function should not be changed. It is checking only for IFF_UP and doesn't care whether the interface is actually operational. This would be a huge behavior change.

Reply:
you are correct. Fixed